core/src/main/java/org/elasticsearch/action/ingest/SimulatePipelineRequest.java

```diff
@@ -135,15 +136,34 @@ static Parsed parseWithPipelineId(String pipelineId, Map<String, Object> config,
         if (pipeline == null) {
             throw new IllegalArgumentException("pipeline [" + pipelineId + "] does not exist");
         }
-        List<IngestDocument> ingestDocumentList = parseDocs(config);
-        return new Parsed(pipeline, ingestDocumentList, verbose);
+        boolean success = false;
+        try {
+            List<IngestDocument> ingestDocumentList = parseDocs(config);
+            Parsed parsed = new Parsed(pipeline, ingestDocumentList, verbose);
+            success = true;
+            return parsed;
+        } finally {
+            if (success == false) {
+                IOUtils.closeWhileHandlingException(pipeline);
```

Comment:
And regardless of if the parsing exception is important, if closing is necessary, failure to close should be added as suppressed exceptions to the parse failure.